### ISSUE TITLE
Downloading STOPS

### ISSUE THREAD

**Issue by ganggreen777** [kzahel/jstorrent-legacy-app#94]
This has been ongoing since I started using this app.

I have to constantly stop and restart a torrent.  Is there a glitch/ problem that is known or what.  The download will get to about 14 or 15% and stop.  Like the peers stalled and stopped downloading.  I have to stop and restart......again and again....
Is there a fix???
Can someone please respond with a valiant effort to help and not a non helpful response

**Comment by kzahel** [comment]
How big is the torrent? How many MB/GB before it stalls?

**Comment by ganggreen777** [comment]
It actually doesnt matter, ive had to do the same on anything i download.
 I did a music torrent at 313mb had to do the same, another music torrent at 125 mb
THis movie torrent is 2.26 gigs

**Comment by kzahel** [comment]
That is very strange. Most users have no problems like this. Are you running from source on github? Are you using ChromeOS?
Perhaps I could inspect what's going on on your setup if you want, you could send me a remote assistance invite to support@jstorrent.com

**Comment by ganggreen777** [comment]
I am running your app on chrome book Toshiba. ....

**Comment by ganggreen777** [comment]
Running as an app from webstore from chrome os
Not sure what meant by github source

**Comment by ganggreen777** [comment]
Are we done discussing the problems.....
bc
I dont think we have a solution
PLEASE.....im not trying to be ugly (sorry if it sounds like it) but I would love to find a problem and solve it.
I am open to remote access to see if you see obviously something causing the problem

**Comment by kzahel** [comment]
Cool can you send me an email to support@jstorrent.com and we'll setup a
time. I really want to find out what's going wrong!

**Comment by ganggreen777** [comment]
Ok with time....let me figure a couple of things....do you have access to
the whole chromebook??? like tech support

**Comment by kzahel** [comment]
Yup! You have to install this: https://chrome.google.com/webstore/detail/chrome-remote-desktop/gbchcmhmhahfdphkhkmpfmihenigjmpp?hl=en

and then you can send me the invite code.
I'm doing my taxes this Mon-Tues but think I should be free Wed (US Pacific time)

**Comment by ganggreen777** [comment]
ok perhaps wed or thurs  us east time

**Comment by themightytadpole1** [comment]
did you resolve the issue as i have the same problem with the same chromebook

**Comment by kzahel** [comment]
Where are you saving the files? On chromebook Downloads folder? Or external drive (like usb or SD card)

**Comment by themightytadpole1** [comment]
in the chromebook download folder

**Comment by kzahel** [comment]
Chromebook encrypts all files in the Downloads folder, so sometimes it can take a while to save files there, and it can cause errors. Try and only download one torrent at a time, and don't use the system resources too heavily. Let me know if that doesn't help...

**Comment by themightytadpole1** [comment]
the files start immediately but soon run out of puff after 5-10 min have increased the peers from the default of 18 to 900 which made a big difference to the speed,if only it can last long enough to more than 10% of the files completed without shutting down js torrents and restarting which does work for a short time.i will give your advice a go and see if it makes any difference-thanks

**Comment by kzahel** [comment]
Also such a high number for peers could also result in blocking the main thread possibly and causing a timeout for writing the data to disk. It might help to reduce that number to a lower amount say 100 or so.

**Comment by themightytadpole1** [comment]
done-thanks

**Comment by ganggreen777** [comment]
hey sorry.....didnt see the email come through......

like 'themighttadepole' said ....yes the download runs out of gas....if you
will.  I changed the number of peers to 150 ( i did have 500) I also
updated my wifi router per bigRED but all I get is 3bps/sec  best i have
with the provider.
I have seen a little better progress, but the process dies down, and the
ETA goes to infinity bc i would assume the peers dies out....
saying all of this, I still have no idea why the heck this is happennig

**Comment by kzahel** [comment]
How many peers in the swarm (look in the Swarm tab and see how many appear, or look at the number in Trackers or next to the Torrent)

Perhaps there just aren't many peers?

Sometimes it can help to try using a magnet link or a .torrent file, switching from one to the other, to get a different set of peers...

**Comment by ganggreen777** [comment]
i save the downloads to sd card

**Comment by ganggreen777** [comment]
i first download a torrent file...then i download the file from that
torrent file....
the things i download have alot of seeders....(is that the same as peers)

**Comment by kzahel** [comment]
Another thing that can definitely help is to make sure you're close to your wifi router! A bad wifi connection can significantly slow down speeds.

**Comment by ganggreen777** [comment]
yea man....the wifi is downstairs as so is my chromebook

**Comment by ganggreen777** [comment]
under swarm

I see alot of net:: ERR_CONNECTION_REFUSED

**Comment by kzahel** [comment]
That's normal. Usually there's a lot of unresponsive peers.

**Comment by ganggreen777** [comment]
So what will be the trick as to the downloading stalling out....stops
I have tried narrowing the peers limit from 500 to about 150.....I have the
best the WiFi offers through my carrier...perhaps Ethernet hook up while
downloading.....
"You tell me"

**Comment by kzahel** [comment]
ganggreen777 and I found out that the torrent was actually completed. When you "Skip" files, the percentage doesn't match correctly with only the skipped files, but the whole torrent. So it looked like the torrent had slowed down because it was basically already done :-)

**Comment by stretch1931** [comment]
Guys, I wanted to chime in here real quick and make sure everyone knows what I just experienced:  I had two torrents being downloaded.  And on one of the torrents, I skipped one file.  The other one was completely untouched.  But just the skipping of this one file, on the one torrent was enough to cause the whole thing to crawl to 50-150 bytes per sec.  As soon as I changed the file from SKIP to NORMAL, both torrents jumped around 50KB/sec to 350KB/sec.

Basically, don't skip anything, regardless of what you need.  Because it probably has a lot to do with this being a sharing network, and how you are denying the other computer the access to the file.  So it kind of denies yours as well... or something of the sort.

**Comment by pepjo** [comment]
My downloads are stopping to, I'm using windows so I don't think it has nothing to do with the encryption problem, nor the internet connection since I have 300Mbps and I use Ethernet to connect to the router.

The downloads start at 3Mb/s (that's ok) but they slowly get down to 500B/s!! and they will eventually stop.

I'm currently downloading two file of 1,5Gb and 1,8Gb, but this happens with any file.

**Comment by kzahel** [comment]
I identified and fixed a bunch of bugs that were causing downloads to stop. Give it a try it's released in the store now.

**Comment by kzahel** [comment]
The legacy Chrome App is no longer maintained. I've rebuilt JSTorrent from scratch — the new version is much more stable. Get it at **[jstorrent.com](https://jstorrent.com)**

If you run into any issues, please leave feedback!
